src/main/java/com/aws/greengrass/mqttclient/AwsIotMqttClient.java

```diff
@@ -165,17 +165,17 @@ long getThrottlingWaitTimeMicros() {
             transactionLimiter.acquire();
             bandwidthLimiter.acquire(message.getPayload().length);
             synchronized (this) {
-                throwIfNotConnected();
+                throwIfNoConnection();
                 logger.atTrace().kv(TOPIC_KEY, message.getTopic()).kv(QOS_KEY, qos.name()).kv("retain", retain)
                         .log("Publishing message");
                 return connection.publish(message, qos, retain);
             }
         });
     }
 
-    private void throwIfNotConnected() {
-        if (!connected()) {
-            throw new MqttException("Client is not connected");
+    private void throwIfNoConnection() {
+        if (connection == null) {
```

Comment:
Why do we make this change?

Reply:
Because we do want to attempt to send even if we think we might be offline. We'll just get a future back which will be completed eventually. I saw this error thrown in a UAT but I cannot tell the reason for the error (since the connection could have been null or we thought we were disconnected). So I realized that even if we think we're not connected, there's no harm in creating the request anyway.